### PR TITLE
String#starts_with? and String#ends_with? with a splat parameter

### PR DESCRIPTION
**Pull request by mamantoha**
This feature makes `String#starts_with?` and `String#ends_with?` more consistent with Ruby's implementations of [start_with?](https://apidock.com/ruby/String/start_with%3F) and [end_with?](https://apidock.com/ruby/v2_5_5/String/end_with%3F) which accepts a list of arguments.

### CODE REVIEW

**Comment by asterite** [comment]
In Ruby this takes a splat, so I think in Crystal it should be the same.

**Comment by mamantoha** [comment]
@asterite agreed. Changed to a splat parameter.

**Comment by HertzDevil** [comment]
What is the use case? Apart from being similar to Ruby, why do you think this is better than the caller calling `any?` directly?

**Comment by mamantoha** [comment]
@HertzDevil I main purpose of this PR is to make it similar to the Ruby implementation.

TL;DR, I mainly use Ruby, and when I today wrote code with Crystal I noticed that `String#ends_with?` don't accept a list of arguments(Array). So I decided to create this PR. Because why not 🤗.

**Comment by asterite** [comment]
I also agree with @HertzDevil here. Also my initial reaction was "is this doing an `and` or an `or`"? Maybe a silly question to ask because `and` wouldn't make sense. But still, I don't know what's the point of accepting multiple arguments vs. doing `any?`.

It's good to copy Ruby things when they make sense, but otherwise...

**Comment by HertzDevil** [comment]
It also provides no performance improvements over calling `any?` directly at the call site. Yet if an improvement is to be made by e.g. converting `starts_with?("foo", "far")` into `starts_with?(Regex.union("foo", "far"))` or a prefix tree, now the hidden setup cost is much higher. For a compiled language like Crystal this kind of tradeoff shall be made explicit at the call site.

**Comment by mamantoha** [comment]
Got it. Otherwise, it's not needed. Closing.
